huaweicloud/services/er/resource_huaweicloud_er_route_table.go

```diff
@@ -0,0 +1,519 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/er/v3/associations"
+	"github.com/chnsz/golangsdk/openstack/er/v3/propagations"
+	"github.com/chnsz/golangsdk/openstack/er/v3/routes"
+	"github.com/chnsz/golangsdk/openstack/er/v3/routetables"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceRouteTable() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceRouteTableCreate,
+		UpdateContext: resourceRouteTableUpdate,
+		ReadContext:   resourceRouteTableRead,
+		DeleteContext: resourceRouteTableDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceRouteTableImportState,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The region where the ER instance and route table are located.`,
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The ID of the ER instance to which the route table belongs.`,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The name of the route table.`,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
+					validation.StringMatch(regexp.MustCompile("^[\u4e00-\u9fa5\\w.-]*$"), "The name only english and "+
+						"chinese letters, digits, underscore (_), hyphens (-) and dots (.) are allowed."),
+				),
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The description of the ER route table.`,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 255),
+					validation.StringMatch(regexp.MustCompile(`^[^<>]*$`),
+						"The angle brackets (< and >) are not allowed."),
+				),
+			},
+			"tags": common.TagsForceNewSchema(),
+			// Attributes
+			"associations": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        routeTableRelationshipSchemaResource(),
+				Description: `The association configuration of the route table.`,
+			},
+			"propagations": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        routeTableRelationshipSchemaResource(),
+				Description: `The propagation configuration of the route table.`,
+			},
+			"routes": {
```

Comment:
if the route table has many `routes`, the resource will be too huge when showing.
Therefore, I advise that you can add a new data source to display the routes.

Reply:
done.